container/Containerfile

```diff
@@ -0,0 +1,233 @@
+# Install ceph and dependencies, and clean up
+# IMPORTANT: in official builds, use '--squash' build option to keep image as small as possible
+#   keeping run steps separate makes local rebuilds quick, but images are big without squash option
+#===================================================================================================
+
+# Pre-reqs
+RUN dnf install -y --setopt=install_weak_deps=False epel-release jq
+
+# Add NFS-Ganesha repo
+RUN \
+    echo "[ganesha]" > /etc/yum.repos.d/ganesha.repo && \
+    echo "name=ganesha" >> /etc/yum.repos.d/ganesha.repo && \
+    echo "baseurl=http://mirror.centos.org/centos/8-stream/storage/\$basearch/nfsganesha-5/" >> /etc/yum.repos.d/ganesha.repo && \
+    echo "gpgcheck=0" >> /etc/yum.repos.d/ganesha.repo && \
+    echo "enabled=1" >> /etc/yum.repos.d/ganesha.repo
+
+# ISCSI repo
+RUN set -x && \
+    curl -s -L https://shaman.ceph.com/api/repos/tcmu-runner/main/latest/centos/8/repo?arch=$(arch) -o /etc/yum.repos.d/tcmu-runner.repo && \
+    case "${CEPH_VERSION_NAME}" in \
+        main) \
+            curl -s -L https://shaman.ceph.com/api/repos/ceph-iscsi/main/latest/centos/8/repo -o /etc/yum.repos.d/ceph-iscsi.repo ;\
+            ;;\
+        quincy|reef) \
+            curl -s -L https://download.ceph.com/ceph-iscsi/3/rpm/el8/ceph-iscsi.repo -o /etc/yum.repos.d/ceph-iscsi.repo ;\
+            ;;\
+    esac
+
+# Ceph repo
+RUN set -x && \
+    rpm --import 'https://download.ceph.com/keys/release.asc' && \
+    if [ "${CEPH_VERSION_NAME}" == "main" ] || ${IS_DEVEL}; then \
+        ARCH=$(arch); if [ "${ARCH}" == "aarch64" ]; then ARCH="arm64"; fi ;\
+        REPO_URL=$(curl -s "https://shaman.ceph.com/api/search/?project=ceph&distros=centos/8/${ARCH}&flavor=${OSD_FLAVOR}&ref=${CEPH_REF}&sha1=latest" | jq -r .[0].url) ;\
+        IS_RELEASE=0 ;\
+        echo "" > crimson-packages.txt ;\
+        if [ "${OSD_FLAVOR}" == "crimson" ]; then \
+            echo "ceph-crimson-osd${CEPH_PACKAGE_VERSION}" > crimson-packages.txt ;\
+        fi ;\
+    else \
+        IS_RELEASE=1 ;\
+        REPO_URL="http://download.ceph.com/rpm-${CEPH_VERSION_NAME}/el8/" ;\
+        echo "" > crimson-packages.txt ;\
+    fi && \
+    rpm -Uvh "$REPO_URL/noarch/ceph-release-1-${IS_RELEASE}.el8.noarch.rpm"
+
+# Copr repos
+# TODO: afaict, scikit is for mgr-diskpredition-local, asyncssh for cephadm, but why do we
+#       encode these as deps only in container instead of including them as a vendored
+#       submodule in the ceph repo or something to be included with ceph packages?
+#       @tchaikov @torchiaf @adk3798 @ktdreyer?
+#       ref: https://github.com/ceph/ceph-container/pull/1821
+#       note: scikit-learn packages are not included in ubi8 in ceph-container
+RUN \
+    dnf install -y --setopt=install_weak_deps=False dnf-plugins-core && \
+    dnf copr enable -y tchaikov/python-scikit-learn && \
+    dnf copr enable -y tchaikov/python3-asyncssh
```

Comment:
@guits @adk3798 @torchiaf @tchaikov do any of you have any thoughts here?

Reply:
For asyncssh at least, the cephadm mgr module needs it. I think we couldn't find any packaging of it in official repos so Kefu had this copr repo set up. We should maybe host this under the copr of somebody more involved in the project at this point, but we still need it somewhere.

Reply:
Thanks for weighing in. It's not a big issue, but it seemed worth asking since I'm not sure how hard it is to set up and maintain copr repos. Regardless, it'll be something to changed later down the line if changing it makes sense.